common/src/primitives/amount.rs

```diff
@@ -42,6 +42,12 @@ impl From<u128> for Amount {
     }
 }
 
+impl From<Amount> for u128 {
```

Comment:
Is this really necessary?

Reply:
In Transaction generation in the tests, I want to generate an output with a random value between 1 and the sum of all inputs of the transaction. `gen_range` requires to be passed a range object, and making this work with Amount seems... nasty. Perhaps there is a way around this entirely but I don't see it right now.

Reply:
Maybe just put it inside the tests?

Reply:
I can't implement this in the mempool tests (as a trait implemented for `Amount`) because the `val` field of `Amount` is private.

I thought at least I could annotate the above `From` with `cfg(test)`, but it turns out code annotated with `cfg(test)`, is not visible across crates (this is an [open issue](https://github.com/rust-lang/cargo/issues/8379)), so the mempool tests wouldn't be able to use it.

Reply:
Solved the need for this by adding a method in the `amount` module for generating a random amount in a given range.